### PR TITLE
Upgrade packages

### PR DESCRIPTION
**Pull request by timhaines**
Upgrades: prettier lint-staged ember-percy ember-concurrency ember-route-action-helper

Also updates some of the app code for the new prettier formatting rules.
See the changelog at https://github.com/prettier/prettier/releases/tag/1.6.0
The one with the big impact seems to be "Print 3 or more chained calls on multiple lines (#2673) by @azz"

### CODE REVIEW

**Comment by azz** [comment]
The changes from "Print 3 or more chained calls on multiple lines" look mostly beneficial to the legibility of the code to me. How do you feel about it?

**Comment by timhaines** [comment]
@azz Personally I really like it.  👍   Will be curious to hear what @fotinakis thinks.

**Comment by azz** [comment]
Particularly happy with this one:

```patch
-      let docsPath = event.target.href.split(hostWithProtocol + '/docs')[1].split('#')[0].slice(1);
+      let docsPath = event.target.href
+        .split(hostWithProtocol + '/docs')[1]
+        .split('#')[0]
+        .slice(1);
```

File: app/components/snapshot-list.js [hunk 1]
```diff
@@ -118,7 +118,12 @@ export default Ember.Component.extend({
       if (this.get('activeSnapshotId')) {
         let index = this.get('snapshotComponents.length') - 1;
 
-        if (this.get('activeSnapshotId') == this.get('sortedSnapshots').objectAt(index).get('id')) {
+        if (
+          this.get('activeSnapshotId') ==
+          this.get('sortedSnapshots')
+            .objectAt(index)
```
Comment (fotinakis) [inline app/components/snapshot-list.js]: I'd probably rewrite this one to pull this out to a variable, so the if statement is clearer and one line.

**Pull request merged**